### ISSUE TITLE
Loop with branches and 'continue' causes incorrect results

### ISSUE THREAD

**Issue by ma-ba** [numba/numba#1522]
The following function gives incorrect results when compiled with Numba 0.22.0:
```python
from numba import jit


def loop_with_branches_and_continue(x):
    ret = 2
    for _ in range(1):
        a = x
        if a != x:
            if a != x:
                a = x
            continue
        if a == x:
            ret = a
    return ret


def test():
    pyfunc = loop_with_branches_and_continue
    cfunc = jit(nopython=True)(pyfunc)

    x = 1
    expected = pyfunc(x)
    got = cfunc(x)

    assert expected == got, 'expected: {}, got: {}'.format(expected, got)


if __name__ == "__main__":
    test()
```
Can you reproduce this? I tried to minimize the example but did not look at any compiled code, so I have no idea what is causing it.

**Comment by sklam** [comment]
I can reproduce this error.

**Comment by sklam** [comment]
The bug:

There is an erroneous `del a` inserted after the innermost assignment to `a` in the numba IR.

```python
def loop_with_branches_and_continue(x):
    ret = 2
    for _ in range(1):
        a = x
        if a != x:
            if a != x:
                a = x   # `del a` inserted after this line
            continue
        if a == x:
            ret = a
    return ret
```
This is likely to be a bug in the lifetime analysis.  That somehow has a nonlocal effect.

**Comment by ma-ba** [comment]
Looks like this is resolved by #1611.
